content/blog/2022/10/2022-10-10-plugin-health-scoring-system-report.adoc

```diff
@@ -0,0 +1,130 @@
+
+We aim to fulfill this need by introducing a metric system to calculate the health score of Jenkins plugins. This health score measures the plugin’s development maturity level with the help of accurate numeric figures. This score attempts to provide an accurate picture of how much care and help a plugin needs and makes it easier for contributors to contribute. This score also allows users to make a conscious decision before installing and/or using a plugin.
+
+There were two coding phases in the GSoC 2022.
+
+---
+
+== Architecture Diagram
+
+image::/images/gsoc/2022/plugin-health-score/architecture-diagram.png[Architecture Diagram, height=690, role=center, float=center]
+
+---
+
+== Phase 1
+
+In this phase, we worked on setting up the project and followed best practices while doing so. The whole application is dockerized so that it is easier for anyone to spin up this project and test it out locally. There's continuous integration (CI) in place for the project to make sure that new changes are tested automatically. The useful details from the Jenkins Update Center are stored inside the database to run analysis.
+
+=== Deliverables
+
+* Create a spring boot project connected with a PostgreSQL DB (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/3[PR])
+* Dockerize the whole application (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/7[PR])
+* Setup continuous integration for the project (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/10[PR])
+* Store plugin details from the update center, inside the DB in JSON format (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/18[PR])
+
+=== Resources
+
+* link:https://github.com/jenkins-infra/plugin-health-scoring/milestone/1?closed=1[View Milestone 1]
+* Jenkins Online Meetup Phase 1: link:https://docs.google.com/presentation/d/1t2vuNn1NFpDusnw0m4vdFw6WBQMeU6kccv_K1v2L6R0/edit#slide=id.g13dcaed2105_0_0[Slides]
+* Jenkins Online Meetup Phase 1: Demo and Presentation
+
+video::loLSNdCv6K4[youtube,width=800,height=420,start=1089]
+
+---
+
+== Phase 2
+
+In this phase, the probe engine for the probes was designed. And processes like reading the update center and executing the probe engine are now running on a schedule without the need to restart the application. There are additions of useful probes running in the probe engine, which are getting listed on the UI to highlight the current state of the project.
+
+=== Deliverables
+
+* Design the Probe engine for the probes (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/19[PR])
+* Schedule the reading of the update center, and probe engine execution (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/20[PR#1] and link:https://github.com/jenkins-infra/plugin-health-scoring/pull/30[PR#2])
```

Comment:
```suggestion
* Schedule the reading of the update center and probe engine execution (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/20[PR#1] and link:https://github.com/jenkins-infra/plugin-health-scoring/pull/30[PR#2])
```
removing the "," would be okay here as it is not needed

Reply:
Done, thanks!